src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionAttachmentManager.cs

```diff
@@ -312,9 +314,12 @@ private void AddNewFileTransfer(FileTransferInformation fileTransferInfo, AsyncC
                     {
                         if (t.Exception == null)
                         {
-                        // Uri doesn't recognize file paths in unix. See https://github.com/dotnet/corefx/issues/1745
-                        var attachmentUri = new UriBuilder() { Scheme = "file", Host = "", Path = localFilePath }.Uri;
-                            this.AttachmentSets[fileTransferInfo.Context][uri].Attachments.Add(new UriDataAttachment(attachmentUri, fileTransferInfo.Description));
+                            lock (lockObject)
```

Comment:
Try to use `ConcurrentDictionary` here.

Reply:
Switched to concurrent dictionaries wherever required but cannot remove lock as a lot of code depends on IList